### PR TITLE
515 enable read/write of bool arrays

### PR DESCRIPTION
**Pull request by hokiegeek2**

### CODE REVIEW

File: tests/io_test.py
Comment (reuster986) [inline tests/io_test.py:101]: These `assertEqual` tests should use the form:
```python
self.assertEqual((self.dict_columns['name'] == retrieved_columns['name']).all())
```
In their current form, the tests will not actually compare the values of the arrays.

File: tests/io_test.py
Comment (reuster986) [inline tests/io_test.py:123]: same issue with this block of tests

File: tests/io_test.py
Comment (reuster986) [inline tests/io_test.py:216]: same issue here

File: tests/io_test.py
Comment (reuster986) [inline tests/io_test.py:241]: same issue

File: src/GenSymIO.chpl
Comment (reuster986) [inline src/GenSymIO.chpl:479]: You can get rid of a copy here by deleting line 477 and changing 479 to:

```chpl
entryBool.a = entryInt.a:bool;
```

**Review by reuster986 — CHANGES_REQUESTED** [review]
A few issues with the tests, but I think the logic looks good.

File: src/GenSymIO.chpl
Comment (hokiegeek2) [reply src/GenSymIO.chpl]: Good point, fixed and pushed to git

Comment (hokiegeek2) [reply src/GenSymIO.chpl]: Good point, fixed and pushed to git

File: tests/io_test.py
Comment (hokiegeek2) [reply tests/io_test.py]: fixed and pushed to git

File: tests/io_test.py
Comment (hokiegeek2) [reply tests/io_test.py]: fixed and pushed to git

File: tests/io_test.py
Comment (hokiegeek2) [reply tests/io_test.py]: fixed and pushed to git

File: tests/io_test.py
Comment (hokiegeek2) [reply tests/io_test.py:101]: fixed and pushed to git

**Review by mhmerrill — COMMENTED** [review]
things look good to me but I'll defer to Bill for approval and merging

**Review by reuster986 — APPROVED** [review]
looks good!

**Comment by reuster986** [comment]
I'll let the checks finish before merging.

**Pull request merged**
